flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/utils/InMemoryLookupableTableSource.scala

```diff
@@ -73,121 +79,72 @@ class InMemoryLookupableTableSource(
-    *       (11, 1L, "Julian"),
-    *       (22, 2L, "Jark"),
-    *       (33, 3L, "Fabian"))
-    *
-    *     val source = InMemoryLookupableTableSource.builder()
-    *       .data(data)
-    *       .field("age", Types.INT)
-    *       .field("id", Types.LONG)
-    *       .field("name", Types.STRING)
-    *       .enableAsync()
-    *       .build()
-    * }}}
-    */
-  class Builder {
-    private val schema = new mutable.LinkedHashMap[String, TypeInformation[_]]()
-    private var data: List[Product] = _
-    private var asyncEnabled: Boolean = false
-
-    /**
-      * Sets table data for the table source.
-      */
-    def data(data: List[Product]): Builder = {
-      this.data = data
-      this
+    val rowData = data.map { entry =>
+      Row.of((0 until entry.productArity).map(entry.productElement(_).asInstanceOf[Object]): _*)
     }
 
-    /**
-      * Adds a field with the field name and the type information. Required.
-      * This method can be called multiple times. The call order of this method defines
-      * also the order of the fields in a row.
-      *
-      * @param fieldName the field name
-      * @param fieldType the type information of the field
-      */
-    def field(fieldName: String, fieldType: TypeInformation[_]): Builder = {
-      if (schema.contains(fieldName)) {
-        throw new IllegalArgumentException(s"Duplicate field name $fieldName.")
-      }
-      schema += (fieldName -> fieldType)
-      this
-    }
+    val asyncEnabled = dp.getOptionalBoolean("is-async").orElse(false)
 
-    /**
-      * Enables async lookup for the table source
-      */
-    def enableAsync(): Builder = {
-      asyncEnabled = true
-      this
-    }
+    val bounded = dp.getOptionalBoolean("is-bounded").orElse(false)
 
-    /**
-      * Apply the current values and constructs a newly-created [[InMemoryLookupableTableSource]].
-      *
-      * @return a newly-created [[InMemoryLookupableTableSource]].
-      */
-    def build(): InMemoryLookupableTableSource = {
-      val fieldNames = schema.keys.toArray
-      val fieldTypes = schema.values.toArray
-      Preconditions.checkNotNull(data)
-      // convert
-      val rowData = data.map { entry =>
-        Row.of((0 until entry.productArity).map(entry.productElement(_).asInstanceOf[Object]): _*)
-      }
-      new InMemoryLookupableTableSource(
-        fieldNames,
-        fieldTypes,
-        rowData,
-        asyncEnabled
-      )
-    }
+    new InMemoryLookupableTableSource(tableSchema, rowData, asyncEnabled, bounded)
+  }
+
+  override def requiredContext(): util.Map[String, String] = {
+    val context = new util.HashMap[String, String]()
+    context.put(CONNECTOR_TYPE, "InMemoryLookupableTable")
+    context
+  }
+
+  override def supportedProperties(): util.List[String] = {
+    val supported = new util.ArrayList[String]()
+    supported.add("*")
+    supported
+  }
+}
+
+object InMemoryLookupableTableSource {
+
+  def createTemporaryTable(
```

Comment:
Using `createTemporaryTable` doesn't mean that it will return something. See `TableEnvironment.createTemporaryView` or `ConnectorTableDescriptor.createTemporaryTable`. I think we should use an unify naming.